Add documentation for API key expiration feature (#463)
rubygems/guides: api-key-scopes.md

![Settings API key](/images/api-keys-index.png){:class="t-img"}

> Enter a name to help you identify the environment the API key may be used in (eg: ci-push-key, mirror-webhook-key, etc.). Check all the scopes you may want to enable. You can also optionally set an [expiration date](#set-an-expiration-date) for the key. Click on create when done.

Reviewer: ```suggestion
Enter a name to help you identify the environment the API key may be used in (eg: ci-push-key, mirror-webhook-key, etc.). Check all the scopes you may want to enable. You can also optionally set an [expiration date](#set-an-expiration-date) for the key. Click create when done.
```

![API key created](/images/api-key-created.png){:class="t-img"}

> The *Age* column shows how old is the key. The *Expiration* column shows when the key will expire (if an expiration was set). The *Last access* column shows the last time (in UTC) the key was used in a successful authentication. You can use the **Edit** button to update the scopes of the key. You can use the **Reset** button in the last row to delete *all* the API keys associated with your account.

Reviewer: ```suggestion
The *Age* column shows how old the key is, the *Expiration* column shows when the key will expire (if an expiration was set), and the *Last access* column shows the last time (in UTC) the key was used in a successful authentication. You can use the **Edit** button to update the scopes of the key. You can use the **Reset** button in the last row to delete *all* the API keys associated with your account.
```

Figured it would be nice all in one sentence.

**Note:** When your ownership to a gem is removed, API keys scoped to that gem will become invalid and cannot be used.

Set an expiration date
-----------------------

> When creating an API key, you can optionally set an **expiration date** using the expiration field on the form. Once the expiration date passes, the key becomes invalid and can no longer be used for authentication.

Reviewer: ```suggestion
When creating an API key, we highly recommend you set an **expiration date** using the expiration field on the form. Once the expiration date passes, the key becomes invalid and can no longer be used for authentication.
```

Good place to nudge the user to set one 😄

Author: Good tactics!

![New API key with expiration](/images/new-api-key-expiration.png){:class="t-img"}

Setting an expiration is useful for:

* Short-lived keys used in CI/CD pipelines
* Temporary keys for managing trusted publishing

> * Keys shared with collaborators that should auto-expire

Reviewer: The use cases here aren't realistic. Short lived keys in CI/CD pipelines aren't a great idea since those pipelines will break when they expire, and you shouldn't need a key to manage [trusted publishing](https://guides.rubygems.org/trusted-publishing/).

Setting an expiration is useful for temporary access to make changes (pushing a gem, adding ownership, etc).
